### ISSUE TITLE
Customize `ProblemDetails` result of `NotFound()` call?

### ISSUE THREAD

**Issue by aureole82** [aspnet/AspNetCore#17035]
### Is your feature request related to a problem? Please describe.
Starting with an ASP Core 3.0 API:
I return the result of `NotFound()` in my action.
```cs
[HttpGet("{id}")]
public async Task<IActionResult> Get(int id)
{
    var customer = await _db.GetCustomerAsync(id);

    if (customer == null)
        return NotFound();

    return Ok(customer);
}
```
The `NotFoundResult` is serialized to
```json
{
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.4",
    "title": "Not Found",
    "status": 404,
    "traceId": "|b0f4249b-47f338d62c97e9a9."
}
```
### Describe the solution you'd like
I'd like to comply with the RFC 7807 + pass some more detailed information to the user, maybe:
Returning `NotFound($"Customer#{id} not found.")` results in
```json
{
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.4",
    "title": "Not Found",
    "detail": "Customer#123 not found.",
    "status": 404,
    "traceId": "|b0f4249b-47f338d62c97e9a9."
}
```

Is that or something similar possible?

**Comment by pranavkm** [comment]
@aureole82 you could return a `Problem` instance for this:

```C#
if (customer == null) { return Problem(statusCode: 404, detail: "Customer#123 not found."); }
```

**Comment by aureole82** [comment]
@pranavkm Your answer is wrong. **It always returns HTTP 200 OK!**

It's far more verbose:
```cs
var problem = Problem("Customer#123 not found.", statusCode: (int) HttpStatusCode.NotFound);
problem.StatusCode = (int) HttpStatusCode.NotFound; // DRY?
return problem;
```

**Comment by pranavkm** [comment]
@aureole82 thanks. That's a bug that we addressed in 3.1.0: https://github.com/aspnet/AspNetCore/issues/14663.
